src/zod.ts

```diff
@@ -0,0 +1,231 @@
+/**
+ * Zod validation schemas for config.js.
+ *
+ * All validate functions accept value for configs setting
+ */
+
+import { z } from "zod";
+import { Action, LinkType, MatchMode } from "./constants.js";
+import { existsSync } from "fs";
+
+/**
+ * VALIDATION_SCHEMA is just a object of all the zod schemas
+ * each are named after what they are intended to validate
+ */
+export const VALIDATION_SCHEMA = {
+	URL: z.union([z.string(), z.array(z.string())]).refine(
+		(values) => {
+			const validateUrl = (url) => {
+				try {
+					new URL(url);
+					return true;
+				} catch (error) {
+					return false;
+				}
+			};
+
+			if (Array.isArray(values)) {
+				return values.every(
+					(url) => typeof url === "string" && validateUrl(url)
+				);
+			} else {
+				return typeof values === "string" && validateUrl(values);
+			}
+		},
+		{
+			message: "invalid URL",
+		}
+	),
+	PATH: z.union([z.string(), z.array(z.string())]).refine(
+		(values) => {
+			const validatePath = (value) => {
+				if (typeof value !== "string" || !existsSync(value)) {
+					return false;
+				}
+				return true;
+			};
+
+			if (Array.isArray(values)) {
+				return values.every(validatePath);
+			} else {
+				return validatePath(values);
+			}
+		},
+		{
+			message: "one or more paths do not exist on the filesystem",
+		}
+	),
+	BOOLEAN: z
+		.boolean()
+		.refine((value) => value !== undefined && value !== null, {
+			message: "value must be a defined boolean (true or false)",
+		}),
+	PORT: z.number().refine((value) => value >= 1 && value <= 65535, {
+		message: "port must be a number between 1 and 65535",
+	}),
+	NUMBER: z.union([z.number(), z.undefined()]).refine(
+		(value) => {
+			return typeof value === "number" || value === undefined;
+		},
+		{
+			message: "value must be a number or undefined",
+		}
+	),
+	DELAY: z.number().refine((value) => value >= 1 && value <= 1000, {
+		message:
+			"delay is in seconds, >1000 implies you think it's milliseconds",
+	}),
+	ACTION: z
+		.string()
+		.refine((value) => value === Action.SAVE || value === Action.INJECT, {
+			message: `value must be either '${Action.SAVE}' or '${Action.INJECT}'`,
+		}),
+	MATCHMODE: z
+		.enum([MatchMode.SAFE, MatchMode.RISKY])
+		.refine(
+			(value) => value === MatchMode.RISKY || value === MatchMode.SAFE,
+			{
+				message: `value must be either '${MatchMode.RISKY}' or '${MatchMode.SAFE}'`,
+			}
+		),
+	LINKTYPE: z
+		.enum([LinkType.SYMLINK, LinkType.HARDLINK])
+		.refine(
+			(value) =>
+				value === LinkType.HARDLINK || value === LinkType.SYMLINK,
+			{
+				message: `value must be either '${LinkType.HARDLINK}' or '${LinkType.SYMLINK}'`,
+			}
+		),
+	STRING: z
+		.string()
+		.min(1, { message: "value must have at least 1 character" })
+		.max(24, { message: "value can have a max of 24 characters" }),
+	FUZZY: z.number().refine((value) => value > 0 && value < 1, {
```

Comment:
If it's a decimal percentage, let's call this `DECIMAL_PERCENTAGE`, not `FUZZY`. Also use native checks.

Reply:
Sounds reasonable, I just named it that since it's only use was for fuzzyThreshold currently.